### PR TITLE
chore: upgrade deps and use c8/prettier

### PR DESCRIPTION
**Pull request by 43081j**
* added prettier
* added c8
* updated eslint
* synchronised publish flow with other chai repos

can you take a look @koddsson ?

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "bracketSpacing": false,
+  "printWidth": 80,
+  "semi": true,
+  "singleQuote": true,
+  "tabWidth": 2,
+  "trailingComma": "none",
+  "useTabs": false,
+  "arrowParens": "always"
+}
```
Comment (koddsson) [inline .prettierrc.json:10]: Any reason why we don't just use the defaults from prettier? Also is this consistent with other chai projects?

File: .editorconfig [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
-trim_trailing_whitespace = true
-charset = utf-8
-indent_style = space
-indent_size = 4
+root = true
+
+[*]
+end_of_line = lf
+indent_size = 2
+indent_style = space
```
Comment (koddsson) [inline .editorconfig:6]: If we're changing this, we could change `indent_style` to `tab` and then setting the tab width is just up to each person via their editor settings. 

What do you think?

**Review by koddsson — APPROVED** [review]

**Comment by koddsson** [comment]
I have some opinions but I don't hold them strong enough to halt this if you feel stronger about it so it's up to you :)

File: .prettierrc.json (hunk 1 above)
Comment (43081j) [reply .prettierrc.json:10]: this is the same as the one in other chai projects

File: .editorconfig (hunk 2 above)
Comment (43081j) [reply .editorconfig:6]: i think given this is the same in other chai repos, we should probably do that across the board in a standalone PR

**Pull request merged**